ql/src/java/org/apache/hadoop/hive/ql/parse/ReplicationSpec.java

```diff
@@ -437,4 +448,30 @@ public void setNeedDupCopyCheck(boolean isFirstIncPending) {
     // Check HIVE-21197 for more detail.
     this.needDupCopyCheck = isFirstIncPending;
   }
+
+  public boolean isPathOwnedByHive() {
+    return isPathOwnedByHive;
+  }
+
+  public static boolean isPathOwnedByHive(HiveConf conf, String user) {
```

Comment:
Can this method moved to ReplUtils as it is static method. ReplicationSpec is shared by both source and target but this method is needed only in source.

Reply:
This method is used at target during truncate ...truncate is not using the replication scope from disk ...